src/main/java/org/opentripplanner/routing/algorithm/RoutingWorker.java

```diff
@@ -148,29 +164,59 @@ public RoutingResponse route(Router router) {
+            request,
+            StreetMode.FLEXIBLE.equals(request.modes.egressMode)
+                ? StreetMode.WALK
+                : request.modes.egressMode,
+            true,
+            2000
+        );
 
         AccessEgressMapper accessEgressMapper = new AccessEgressMapper(transitLayer.getStopIndex());
-        Collection<AccessEgress> accessTransfers = accessEgressMapper.mapNearbyStops(accessStops, false);
-        Collection<AccessEgress> egressTransfers = accessEgressMapper.mapNearbyStops(egressStops, true);
+        Collection<AccessEgress> accessTransfers = new ArrayList<>();
+        Collection<AccessEgress> egressTransfers = new ArrayList<>();
 
         List<Itinerary> itineraries = new ArrayList<>();
 
-        if (OTPFeature.FlexRouting.isOn() && request.modes.transitModes.contains(TransitMode.FLEXIBLE)) {
-            FlexRouter flexRouter = new FlexRouter(
-                request.rctx.graph,
-                request.getDateTime().toInstant(),
-                request.arriveBy,
-                ADDITIONAL_SEARCH_DAYS_BEFORE_TODAY,
-                ADDITIONAL_SEARCH_DAYS_AFTER_TODAY,
-                accessStops,
-                egressStops
-            );
+        // Add non-flex access transfers
+        if (!StreetMode.FLEXIBLE.equals(request.modes.accessMode)) {
+            accessTransfers.addAll(accessEgressMapper.mapNearbyStops(accessStops, false));
+        }
+        // Add non-flex egress transfers
+        if (!StreetMode.FLEXIBLE.equals(request.modes.egressMode)) {
+            egressTransfers.addAll(accessEgressMapper.mapNearbyStops(egressStops, true));
+        }
 
-            itineraries.addAll(flexRouter.createFlexOnlyItineraries());
-            accessTransfers.addAll(accessEgressMapper.mapFlexAccessEgresses(flexRouter.createFlexAccesses()));
-            egressTransfers.addAll(accessEgressMapper.mapFlexAccessEgresses(flexRouter.createFlexEgresses()));
+        if (OTPFeature.FlexRouting.isOn()) {
+            if (request.modes.contains(StreetMode.FLEXIBLE)) {
+                FlexRouter flexRouter = new FlexRouter(
+                    request.rctx.graph,
+                    request.getDateTime().toInstant(),
+                    request.arriveBy,
+                    ADDITIONAL_SEARCH_DAYS_BEFORE_TODAY,
+                    ADDITIONAL_SEARCH_DAYS_AFTER_TODAY,
+                    accessStops,
+                    egressStops
+                );
+
+                // Add flex access transfers
+                if (StreetMode.FLEXIBLE.equals(request.modes.accessMode)) {
+                    accessTransfers.addAll(accessEgressMapper.mapFlexAccessEgresses(flexRouter.createFlexAccesses()));
+                }
+                // Add flex egress transfers
+                if (StreetMode.FLEXIBLE.equals(request.modes.egressMode)) {
+                    egressTransfers.addAll(accessEgressMapper.mapFlexAccessEgresses(flexRouter.createFlexEgresses()));
+                }
+            }
```

Comment:
The `routeTransit` should give an overview of the transit routing - and delegate as much of the business logic as possible - the FLEX now complicate to main logic a lot and if we introduced another similar concept then it would become hard to understand. Also, people who do not care about FLEX should be able to skip all flex code. The new code in this method is not at the same abstraction level and it need to refactored out. Some of it should maybe be pushed to other classes and some into methods in this class. Enums are compared by `equals`. Also, the flow is very hard to follow, the check on mode is/is not FLEX is probably a business concept (not DRY) witch should have a name.

Reply:
I have refactored out some of the access/egress logic for flex. I had to duplicate some code, as the flex access/egress search requires a regular access/egress search as input. It might be that a few lines of duplicated code is better than combining the logic for flex access/egress and regular access/egress.

I still does look somewhat complicated, because you need to perform a check on access and egress separately, to see if you should use the flex logic or the regular logic. Still, it is more readable than what it was before.